### PR TITLE
Add input endpoint management to Azure

### PR DESCRIPTION
**Pull request by techhat**
The Azure SDK does not support input endpoint management, so we use the Azure REST API directly here.

### CODE REVIEW

File: salt/cloud/clouds/msazure.py [hunk 1]
```diff
@@ -1788,6 +1792,272 @@ def delete_management_certificate(kwargs=None, conn=None, call=None):
...
+    </ConfigurationSet>
+  </ConfigurationSets>
+  <OSVirtualHardDisk>
+  </OSVirtualHardDisk>
+</PersistentVMRole>'''.format(endpoints_xml)
+
+    path = 'services/hostedservices/{0}/deployments/{1}/roles/{2}'.format(
+        kwargs['service'],
+        kwargs['deployment'],
+        'techhatmaster',
```
Comment (s0undt3ch) [inline salt/cloud/clouds/msazure.py]: techhatmaster?

Comment (techhat) [inline salt/cloud/clouds/msazure.py]: Gah! How embarrassing. Fixing now.

**Comment by traxair** [comment]
This was a must-have feature! Thanks @techhat

File: salt/cloud/clouds/msazure.py (hunk 1 above)
Comment (techhat) [inline salt/cloud/clouds/msazure.py]: Fixed in #21693.

**Pull request merged**